fetch_mail.py

```diff
@@ -0,0 +1,160 @@
+        -> save into a file
+        -> quit()
+"""
+
+import imaplib  # For reading messages
+import logging  # For catching logs
+import email
+import re
+import email.policy
+import socket  # Require for imaplib. connection
+
+
+class LPForum:
+    def __init__(self, email, password):
+        self.email_user = email
+        self.email_pass = password
+
+    def __str__(self):
+        return str(self.email_user)
+
+    def login(self):
+        """ Creating a secure connection over SSL socket
+        Return an imap4 object to "mail" variable """
+        try:
+            mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+
+            # login to user account
+            mail.login(self.email_user, self.email_pass)
+            return mail
+        except imaplib.IMAP4.error as err:
+            logging.info(err)
+            return False
+
+    def get_message_id(self, obj):
+        """ Method for Return the number of mails
+        specified in the mailbox. """
+
+        obj.select(mailbox="INBOX")  # Select a mailbox first
+        _, msgnums = obj.search(
+            "utf-8", "(Unseen)", "FROM", "linkinpark@discoursemail.com"
+        )  # Search message inside mailbox
+        logging.info(f"Found unseen mail list : {msgnums}")
+        if len(msgnums[0]) > 0:
+            return msgnums
+        return False
+
+    def fetch_raw_message(self, obj, message_id):
+        """ Extracting raw message from message id's.
+        message_id contains id's of email messages
+        by linkinpark@discoursemail. 
+        (RFC822) is a msg. format, defines => header, payload
+        """
+
+        ids = message_id[0].split()
+        data = []
+        for id_ in ids:
+            typ, response = obj.fetch(id_, "(RFC822)")
+            data.append(response[0][1])
+
+        if typ == "OK":
+            # raw_text contains two part -> id[0][0] and message body[0][1]
+            return data
+        else:
+            return False
+
+    def extract_contents(self, raw_data: list):
+        """ Parse the message :
+        header  (recipient information)
+        payload (content) """
+
+        # Get Headers
+        Headers = {}
+        messages_obj = [
+            email.message_from_bytes(raw_, policy=email.policy.default)
+            for raw_ in raw_data
+        ]
+
+        Headers["To"] = messages_obj[0]["To"]
+        Headers["Subject"] = messages_obj[0]["Subject"]
+
+        # Get different headers
+        Headers["From"] = []
+        Headers["Reply-To"] = []
+        for msg in messages_obj:
+            Headers["From"].append(msg["From"])
+            Headers["Reply-To"].append(msg["Reply-To"])
+
+        # Get Payloads
+        payload_body = []
+        for objI in messages_obj:
+            payload_body.append(objI.get_payload()[0].get_content())
+
+        # pattern = r"(\[[\W\w]{9}user_tag.+?(?=\]).+?(?=\[\W\w{5}\]))(.*?(?=\[[\w]{5}))"
+        user_tag = "Himan10"
+        regex_pattern = r"(\[/?quote(?:=[^]]*)?\])"
```

Comment:
Might be good to comment what you are looking for. regex is hard to read.

Reply:
this regex pattern simply extract the message which either starts with [quote= (Opening) or [/quote (Closing)
This "[quote=" is used for quoting someone's message 
For example: suppose if someone quote my reply. so it looks like this ->
```
[quote="Himan10, post:1056, topic:36818"]
i heard from someone that shinoda used to post in here… is it really real ?
[/quote]

No idea, over a decade ago, perhaps.
```
Now inside the function extract_contents(), it first split the entire message according to the regex provided above and then save the message which matches the following pattern -> "[quote]Himan10Message[/quote] Second user reply"